ansible/setup.yml

```diff
@@ -6,14 +6,20 @@
   - name: find the ip of docker-machine
     local_action: shell "docker-machine" "ip" "{{docker_machine_name | default('whisk')}}"
     register: result
-    ignore_errors: true
+    when: "'environments/mac' in inventory_dir"
 
-  - set_fact: 
+  - name: get the docker-machine ip
+    set_fact:
       docker_machine_ip: "{{ result.stdout }}"
-  
+    when: "'environments/mac' in inventory_dir"
+
   - name: gen hosts for docker-machine 
     local_action: template src="{{playbook_dir}}/environments/mac/hosts.j2.ini" dest="{{ playbook_dir }}/environments/mac/hosts"
-    when: result.rc == 0
+    when: "'environments/mac' in inventory_dir"
+
+  - name: add new db host on-the-fly
```

Comment:
line 20 is for a good reason. Because at the beginning, we don't have a hosts file for mac, which is needed for generating the `db_local.ini`. That's why line 20 adds the newly generated db host on-the-fly into ansible environment.